### PR TITLE
[Merged by Bors] - feat: port Topology.ContinuousOn

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/ContinuousOn.lean [hunk 1]
```diff
@@ -149,25 +147,17 @@ theorem nhdsWithin_le_iff {s t : Set α} {x : α} : 𝓝[s] x ≤ 𝓝[t] x ↔
   simp_rw [Filter.le_def, mem_nhdsWithin_iff_eventually]
   constructor
   · exact fun h => (h t <| eventually_of_forall fun x => id).mono fun x => id
-  · exact fun h u hu => (h.And hu).mono fun x hx h => hx.2 <| hx.1 h
+  · exact fun h u hu => (h.and hu).mono fun x hx h => hx.2 <| hx.1 h
 #align nhds_within_le_iff nhdsWithin_le_iff
 
+-- porting note: golfed, droped an unneeded assumption
 theorem preimage_nhdsWithin_coinduced' {π : α → β} {s : Set β} {t : Set α} {a : α} (h : a ∈ t)
```
Comment (urkud) [inline Mathlib/Topology/ContinuousOn.lean:154]: I'm going to backport this.

File: Mathlib/Topology/ContinuousOn.lean [hunk 2]
```diff
@@ -1100,62 +1067,54 @@ theorem Embedding.map_nhdsWithin_eq {f : α → β} (hf : Embedding f) (s : Set
...
-        exact
-          continuousWithinAt_of_not_mem_closure fun h =>
-            this (closure_inter_subset_inter_closure _ _ h).2
-      ·
-        exact
-          (hg x hx).congr (fun y hy => piecewise_eq_of_not_mem _ _ _ hy.2)
-            (piecewise_eq_of_not_mem _ _ _ hx.2)
-#align continuous_on.piecewise' ContinuousOn.piecewise'
-
 theorem ContinuousOn.if' {s : Set α} {p : α → Prop} {f g : α → β} [∀ a, Decidable (p a)]
```
Comment (urkud) [inline Mathlib/Topology/ContinuousOn.lean:1083]: Lean failed to find a `Decidable` instance to apply the `piecewise'` version here, so I swapped the lemmas.

File: Mathlib/Topology/ContinuousOn.lean [hunk 3]
```diff
@@ -306,15 +295,6 @@ theorem nhdsWithin_compl_singleton_sup_pure (a : α) : 𝓝[≠] a ⊔ pure a =
   rw [← nhdsWithin_singleton, ← nhdsWithin_union, compl_union_self, nhdsWithin_univ]
 #align nhds_within_compl_singleton_sup_pure nhdsWithin_compl_singleton_sup_pure
 
-/- ./././Mathport/Syntax/Translate/Expr.lean:177:8: unsupported: ambiguous notation -/
-theorem nhdsWithin_prod_eq {α : Type _} [TopologicalSpace α] {β : Type _} [TopologicalSpace β]
```
Comment (jcommelin) [inline Mathlib/Topology/ContinuousOn.lean]: What happened to this decl?

Comment (urkud) [reply Mathlib/Topology/ContinuousOn.lean]: @jcommelin It was moved to `Topology.Constructions`.

Comment (jcommelin) [reply Mathlib/Topology/ContinuousOn.lean]: Apparently it moved to `Topology.Constructions`.

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**